packages/gatsby/src/redux/persist.ts

```diff
@@ -24,6 +25,10 @@ const getReduxCacheFolder = (): string =>
   // This is a function for the case that somebody does a process.chdir (#19800)
   path.join(process.cwd(), `.cache/redux`)
 
+const getWorkerSlicesFolder = (): string =>
+  // This is a function for the case that somebody does a process.chdir (#19800)
+  path.join(process.cwd(), `.cache/worker`)
```

Comment:
I started to wonder wether we should force clean this dir when we start `gatsby` so we don't keep "stale" state slices data from previous run that might be accidentally loaded by worker? Might not be for this PR (rather separate as this PR is already quite large), but we should also think about ensuring slices loaded in worker are fresh

Reply:
Created a story for that in clubhouse